lib/ansible/module_utils/k8s/raw.py

```diff
@@ -71,6 +75,7 @@ def argspec(self):
         argument_spec['wait_timeout'] = dict(type='int', default=120)
         argument_spec['validate'] = dict(type='dict', default=None, options=self.validate_spec)
         argument_spec['append_hash'] = dict(type='bool', default=False)
+        argument_spec['apply'] = dict(type='bool', default=False)
```

Comment:
I understand defaulting to false for the sake of an unsurprising update, but should we consider making apply the default (in this or in a later version) since it will likely be so much less error prone?

Reply:
I agree with you, I'm just not sure how to do this in an unsurprising backward compatible way.

Reply:
Well, backwards compatibility should remain (using apply vs the old way shouldn't really change anything, other than decreasing the number of potential errors), the only real difference between the old way and `apply` should be that an annotation appears on your object. It might be enough to mention this in the porting guide and just rip the bandaid off, rather than defaulting to an inferior implementation. I'm not sure though, right now it just feels more wrong to add apply and not set it as default than it does to make the change.

Reply:
The problem here is that if we set default apply to true, the module will fail for anyone on older openshift libraries.

I might have to work around that by not setting a default at all, and then assume True if openshift is new enough unless explicitly set False